eofconvention.fsx

```diff
@@ -0,0 +1,29 @@
+open System.IO
+
+let EOFIsEOL(path: FileInfo) = 
+    try
+        use sr = new StreamReader (path.FullName)
+        let filetext = sr.ReadToEnd()
+        (filetext |> Seq.last = '\n')
+        
+    with
+        | :? System.ArgumentException -> printfn "%A" path; true
+        
+
+let EOFIsNotEOL(path: FileInfo) = not (EOFIsEOL path)
+
+let NotPrefix (value: string) (str: string) =
+    not (str.StartsWith(value))
+
+
+let invalidFiles = 
+    Directory.GetFiles(".", "*.*", SearchOption.AllDirectories) 
+    |> Array.filter (NotPrefix "./node_modules")
+    |> Array.filter (NotPrefix "./.git")
+    |> Array.map (fun x -> FileInfo(x))
+    |> Array.filter EOFIsNotEOL
+
+if invalidFiles.Length > 0 then
+    let mutable message = "The following files doesn't end with EOL:\n" 
+    invalidFiles |> Array.iter (fun x -> message <- message + x.FullName + "\n")
+    raise (System.Exception(message))
```

Comment:
@realmarv nit: a more visually-appealing way to write this with less parentheses is `raise <| System.Exception message`; but you could even learn here the function `failwith`, which is used when you don't want to raise a specific type of exception, but just the generic one, so `failwith message`

Reply:
Done